install/common-pre-install/pre-installation-cpi.md

```diff
@@ -0,0 +1,560 @@
+
+### 2.1 Generate topology files
+
+> **Note:** The following seed files are auto-generated with the common pre-installer `application_node_config.yaml` ,`hmn_connections.json` ,`ncn_metadata.csv` ,`switch_metadata.csv`. See [Seed file generation](hpcm_installation-cpi.md#seed-file-generation).
+
+1. Verify if `cabinets.yaml` config file has not been created (manually).
+
+   If `cabinets.yaml` config file has not been created, create the `cabinets.yaml` using the following step, else skip the following step.
+
+      (`pit#`)
+      1. Create remaining seed files,  unless they already exist from a previous installation.
+         - [Create `cabinets.yaml`](../create_cabinets_yaml.md)
+
+1. (`pit#`) Assuming all seed files are under `$HOME/seedfiles` directory, copy the generated files under `${PITDATA}/prep` directory.
+
+   ```bash
+   cp $HOME/seedfiles/* "${PITDATA}/prep"
+   ```
+
+1. (`pit#`) Confirm that the following files exist.
+
+   ```bash
+   ls -l "${PITDATA}"/prep/{application_node_config.yaml,cabinets.yaml,hmn_connections.json,ncn_metadata.csv,switch_metadata.csv}
+   ```
+
+   Expected output look similar to the following example:
+
+   ```text
+   -rw-r--r-- 1 root root  146 Jun  6 00:12 /var/www/ephemeral/prep/application_node_config.yaml
+   -rw-r--r-- 1 root root  392 Jun  6 00:12 /var/www/ephemeral/prep/cabinets.yaml
+   -rwxr-xr-x 1 root root 3768 Jun  6 00:12 /var/www/ephemeral/prep/hmn_connections.json
+   -rw-r--r-- 1 root root 1216 Jun  6 00:12 /var/www/ephemeral/prep/ncn_metadata.csv
+   -rw-r--r-- 1 root root  150 Jun  6 00:12 /var/www/ephemeral/prep/switch_metadata.csv
+   ```
+
+### 2.2 Customize `system_config.yaml`
+
+1. (`pit#`) Create or copy `system_config.yaml`.
+
+   - If one does not exist from a prior installation, then create an empty one:
+
+      ```bash
+      csi config init empty
+      ```
+
+   - Otherwise, copy the existing `system_config.yaml` file into the working directory and proceed to the [Run CSI](#33-run-csi) step.
+
+1. (`pit#`) Edit the `system_config.yaml` file with the appropriate values.
+
+   > **NOTE:**
+   >
+   > - For a short description of each key in the file, run `csi config init --help`.
+   > - For more description of these settings and the default values, see [Default IP Address Ranges](../../introduction/csm_overview.md#2-default-ip-address-ranges) and the other topics in [CSM Overview](../../introduction/csm_overview.md).
+   > - To enable or disable audit logging, refer to [Audit Logs](../../operations/security_and_authentication/Audit_Logs.md) for more information.
+   > - If the system is using a `cabinets.yaml` file, be sure to update the `cabinets-yaml` field with `'cabinets.yaml'` as its value.
+
+   ```bash
+   vim system_config.yaml
+   ```
+
+### 2.3 Run CSI
+
+1. (`pit#`) Generate the initial configuration for CSI.
+
+   This will validate whether the inputs for CSI are correct.
+
+   ```bash
+   csi config init
+   ```
+
+   Expected Output:
+
+   ```text
```

Comment:
In this codeblock, replace the internal system name `gamora` with something more generic, such as `system_name`